tests/mechanic/java_resolver_test.py

```diff
@@ -30,7 +30,9 @@ def test_resolves_java_home_for_default_runtime_jdk(self, resolve_jvm_path):
         cfg = config.Config()
         cfg.add(config.Scope.application, "mechanic", "runtime.jdk", None)
 
-        major, java_home = java_resolver.java_home("12,11,10,9,8", cfg)
+        major, java_home = java_resolver.java_home("12,11,10,9,8",
+                                                   specified_runtime_jdk=cfg.opts("mechanic", "runtime.jdk"),
```

Comment:
There is no need for the `Config` object anymore and therefore we should simplify the test and only pass the actual value here.